apps/prairielearn/elements/pl-order-blocks/pl-order-blocks.js

```diff
@@ -3,11 +3,113 @@
+  const maxIndent = options.maxIndent; // defines the maximum number of times an answer block can be indented
+  const enableIndentation = options.enableIndentation;
 
-  let optionsElementId = '#order-blocks-options-' + uuid;
-  let dropzoneElementId = '#order-blocks-dropzone-' + uuid;
+  const optionsElementId = '#order-blocks-options-' + uuid;
+  const dropzoneElementId = '#order-blocks-dropzone-' + uuid;
+  const fullContainer = document.querySelector('.pl-order-blocks-question-' + uuid);
+
+  function initializeKeyboardHandling() {
+    const blocks = fullContainer.querySelectorAll('.pl-order-block');
+    blocks.forEach((block) => initializeBlockEvents(block));
+  }
+
+  function inDropzone(block) {
+    const parentArea = block.closest('.pl-order-blocks-connected-sortable');
+    return parentArea.classList.contains('dropzone');
+  }
+
+  function getIndentation(block) {
+    return Math.round(parseInt(block.style.marginLeft.replace('px', '') / TABWIDTH));
+  }
+
+  function setIndentation(block, indentation) {
+    if (indentation >= 0 && indentation <= maxIndent) {
+      block.style.marginLeft = indentation * TABWIDTH + 'px';
+    }
+  }
+
+  function initializeBlockEvents(block) {
+    function removeSelectedAttribute() {
+      block.classList.remove('pl-order-blocks-selected');
+    }
+
+    function handleKey(ev, block, handle) {
+      // When we manipulate the location of the block, the focus is automatically removed by the browser,
+      // so we immediately refocus it. In some browsers, the blur even will still fire in this case even
+      // though we don't want it to, so we temporarily remove and then reattach the blur event listener.
+      block.removeEventListener('blur', removeSelectedAttribute);
+      handle();
+      ev.preventDefault();
+      block.addEventListener('blur', removeSelectedAttribute);
+      block.focus();
+      setAnswer();
+    }
+
+    function handleKeyPress(ev) {
+      if (!block.classList.contains('pl-order-blocks-selected')) {
+        if (ev.key === 'Enter') {
+          handleKey(ev, block, () => block.classList.add('pl-order-blocks-selected'));
+        }
+      } else {
+        switch (ev.key) {
+          case 'ArrowDown':
+            handleKey(ev, block, () => {
+              if (block.nextElementSibling) {
+                block.nextElementSibling.insertAdjacentElement('afterend', block);
+              }
+            });
+            break;
+          case 'ArrowUp':
+            handleKey(ev, block, () => {
+              if (block.previousElementSibling) {
+                block.previousElementSibling.insertAdjacentElement('beforebegin', block);
+              }
+            });
+            break;
+          case 'Delete':
+          case 'Backspace':
+            handleKey(ev, block, () => {
+              if (inDropzone(block)) {
+                block.style.marginLeft = '0px';
+              }
+              $(optionsElementId)[0].insertAdjacentElement('beforeend', block);
```

Comment:
You should be able to use vanilla JS here (same below):
```suggestion
              document.querySelector(optionsElementId).insertAdjacentElement('beforeend', block);
```